LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -1061,5 +1961,118 @@ private static StructProperty GenerateExpressionGUID()
 
             return new StructProperty("Guid", props, "ExpressionGUID", true);
         }
+
+        private static void ShowError(string errMsg)
+        {
+            MessageBox.Show(errMsg, "Warning", MessageBoxButton.OK);
+        }
+
+        /// <summary>
+        /// Prompts the user for an int, verifying that the int is valid.
+        /// </summary>
+        /// <param name="msg">Message to display for the prompt.</param>
+        /// <param name="err">Error message to display.</param>
+        /// <param name="biggerThan">Number the input must be bigger than. If not provided -2,147,483,648 will be used.</param>
+        /// <param name="title">Title for the prompt.</param>
+        /// <returns>The input int.</returns>
+        private static int promptForInt(string msg, string err, int biggerThan = -2147483648, string title = "")
+        {
+            if (PromptDialog.Prompt(null, msg, title) is string stringPrompt)
+            {
+                int intPrompt;
+                if (string.IsNullOrEmpty(stringPrompt) || !int.TryParse(stringPrompt, out intPrompt) || !(intPrompt > biggerThan))
+                {
+                    MessageBox.Show(err, "Warning", MessageBoxButton.OK);
+                    return -1;
+                }
+                return intPrompt;
+            }
+            return -1;
+        }
+
+        /// <summary>
+        /// Gets a union of two strings. Assumes both only difer at the end.
+        /// </summary>
+        /// <param name="s1">First string.</param>
+        /// <param name="s2">Second string.</param>
+        /// <returns>Union of input strings.</returns>
+        private static string GetUnionOfStrings(string s1, string s2)
+        {
+            // Get the old name found in all pieces of the conversation by getting the union of the bioconversation
+            // and the wwise bank names.
+            // Assumes that both begin the same, since that's the behavior in all vanilla occurences, and helps
+            // keep the logic simple.
+            string union = "";
+            for (int w = 0, b = 0; w < s1.Length && b < s2.Length; w++, b++)
+            {
+                if (char.ToLower(s1[w]) == char.ToLower(s2[b]))
+                {
+                    union += char.ToLower(s1[w]);
+                }
+                else
+                {
+                    break;
+                }
+            }
+            return union;
```

Comment:
building a string character by character is inefficient, as a new string must be allocated on every append. Better to create a new string in one go, by slicing the existing one.
```suggestion
            if(s1.Length == 0 || s2.Length == 0 || char.ToLower(s1[0]) == char.ToLower(s2[0]))
            {
                return "";
            }
            for(int i = 1; i < s1.Length && i< s2.Length; i++)
            {
                if (char.ToLower(s1[i]) != char.ToLower(s2[i]))
                {
                    return s1[..i];
                }
            }
            return s1.Length < s2.Length ? s1 : s2;
```

Reply:
Shouldn't it be `char.ToLower(s1[0]) != char.ToLower(s2[0])`?

Reply:
Whoops! You are correct.